platform/storage/posix/src/mender-storage.c

```diff
@@ -151,24 +151,101 @@ mender_storage_delete_authentication_keys(void) {
 }
 
 mender_err_t
-mender_storage_set_deployment_data(char *deployment_data) {
+mender_storage_set_deployment_data(cJSON *deployment_data) {
     assert(NULL != deployment_data);
-    size_t deployment_data_length = strlen(deployment_data);
 
-    if (MENDER_OK != mender_storage_write_file(MENDER_STORAGE_NVS_DEPLOYMENT_DATA, deployment_data, deployment_data_length)) {
+    /* Add/update version number. We always want to store the deployment data in
+     * the format of the latest and greatest version. By updating the version
+     * number here and validating the data below, we make sure to enforce this.
+     */
```

Comment:
Right now `mender_storage_set_deployment_data` is only called when rebooting, while for every other state transition we do `mender_storage_save_update_state`. As we have plans to merge them, let me comment a concern:

If we blindly update the version (when saving the update_state) then we have issues like:
* device uses v1
* reboots into a newer Mender that upgrades the data to v2 in the first state transition
* An error occurs, a rollback is needed
* reboots into the old Mender that does not understand the data

It is not a fatal issue because we would just clean up the deployment and no inform the backend, but the device would be alive.

However, because dealing with migrations is complex, I would for now make the code to only understand one version, no upgrade no nothing. If it has a different version than the one we support, we fail the update at any time. Whenever we have to introduce a v2 then we plan a proper migration step.